### ISSUE TITLE
Add Configuration for Oauth Auth tokens for self-hosted instances

### ISSUE THREAD

**Issue by LongLiveCHIEF** [ItayRosen/OpenIoT#15]
In order to allow Oauth sign-ups for self-hosted instances, make it easy to configure the Oauth Providers the instance administrator prefers to allow.

**Comment by ItayRosen** [comment]
Currently it's set in the `constants` file (in the API and the platform src files). Do you think it should be configured elsewhere?

**Comment by LongLiveCHIEF** [comment]
Yeah, it needs to be somewhere that isn't managed by version control, and probably even dynamically loaded via the standard config hierarchy (env, file, process). 

Plus they should be able to choose which ones to implement out of the ones that are supported at any given time.

**Comment by LongLiveCHIEF** [comment]
The constants files would still be there, they would just need to load the values from the hierarchy before defining them.

**Comment by ItayRosen** [comment]
Yeah that's probably a good idea. Can the process of defining the constants file happen in the docker process somehow? Maybe I'll create a bash script that walks the user through proper installation of the required constants.

**Comment by LongLiveCHIEF** [comment]
I usually use [nconf](https://www.npmjs.com/package/nconf) or something similar.

Interactive configs are cool, but they put a real hamper on automation and deployment control. You also don't want to make something in the app dependent upon docker.

I'm also thinking that all of the Oauth login providers should be designed so that they're _optional_. The biggest requirement for most who want to self-host is going to be the ability to completely isolate the app from the external internet.

We can make the react component dynamic based on the providers it can find configurations for, if any are found at all, and it will only display buttons for those that are configured.
